visual-js/replace_pkg_version.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Check if jq is installed
+if ! command -v jq &> /dev/null
+then
+    echo "jq could not be found. Please install jq to run this script."
+    exit
+fi
+
+# Check if a JSON file is provided as an argument
+if [ "$#" -ne 1 ]; then
+    echo "Usage: $0 <json-file>"
+    exit 1
+fi
+
+JSON_FILE=$1
+
+# Check if the JSON file exists
+if [ ! -f "$JSON_FILE" ]; then
+    echo "JSON file not found!"
+    exit 1
+fi
+
+# Define the mapping between release names and filenames
+release_names=(
```

Comment:
Why not using associative arrays ? https://www.gnu.org/software/bash/manual/html_node/Arrays.html

It may be slightly less error-prone.

Note: Bash5 is ideal, bash4 is the minimum version needed.

Reply:
I wasn't sure if bash4 is on the runner. Updated.

Reply:
You can go to the safe side by adding this at the top of the file:

```
if [ "${BASH_VERSINFO[0]}" -lt 5 ];then
  echo "Bash5 is required" > /dev/stderr
  exit 1
fi
```